janusgraph-core/src/main/java/org/janusgraph/core/ConfiguredGraphFactory.java

```diff
@@ -0,0 +1,163 @@
+ * {@link JanusGraphManager} reference tracker.
+ */
+public class ConfiguredGraphFactory {
+    /**
+     * Creates a {@link JanusGraph} configuration stored in {@ConfigurationGraphManagament}
+     * configurationGraph and a {@link JanusGraph} graph reference according to the single
+     * Template Configuration  previously created by the {@link ConfigurationManagementGraph} API;
+     * if a configuration exists for this graph already, or a Template Configuration does not exist,
+     * we throw a {@link RuntimeException}; If the Template Configuration does not include its
+     * backend's respective keyspace/table/storage_directory parameter, we set the keyspace/table
+     * to the {@link String} supplied graphName or we append the graphName to the supplied
+     * storage_root parameter.
+     *
+     * @param String graphName
+     *
+     * @return JanusGraph
+     */
+    public static synchronized JanusGraph create(final String graphName) {
+        final ConfigurationManagementGraph configGraphManagement = getConfigGraphManagementInstance();
+
+        final Map<String, Object> graphConfigMap = configGraphManagement.getConfiguration(graphName);
+        Preconditions.checkState(null == graphConfigMap, String.format("Configuration for graph %s already exists.", graphName));
+        final Map<String, Object> templateConfigMap = configGraphManagement.getTemplateConfiguration();
+        Preconditions.checkState(null != templateConfigMap,
+                                "Please create a template Configuration using the ConfigurationManagementGraph#createTemplateConfiguration API.");
+        templateConfigMap.put(ConfigurationManagementGraph.PROPERTY_GRAPH_NAME, graphName);
+        templateConfigMap.put(ConfigurationManagementGraph.PROPERTY_CREATED_USING_TEMPLATE, true);
+
+        final JanusGraph g;
+        try {
+            g = (JanusGraph) JanusGraphManager.getInstance().openGraph(graphName, (String gName) -> {
+                return new StandardJanusGraph(new GraphDatabaseConfiguration(new CommonsConfiguration(new MapConfiguration(templateConfigMap))));
+            });
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+        configGraphManagement.createConfiguration(new MapConfiguration(templateConfigMap));
+        return g;
+    }
+
+    /**
+     * Open a {@link JanusGraph} using a previously created Configuration using the
+     * {@link ConfigurationManagementGraph} API; if a corresponding configuration does not exist, we throw
+     * a {@link RuntimeException}, else return the Graph;
+     *
+     * NOTE: If your configuration corresponding to this graph does not contain information about
+     * the backend's keyspace/table/storage directory, then we set the keyspace/table to the
+     * graphName or set the storage directory to the storage_root + /graphName.
+     */
+    public static JanusGraph open(String graphName) {
```

Comment:
`open` is very specifically attached to an existing previously created configuration for this graph, stored on the `ConfigurationManagementGraph's` graph object as a vertex-- we create the graph according to this specific configuration if it does not exist; and if its does and lives in the cache, we simply return the graph.

`create` is more attached to the creation of the `configuration vertex` on the `ConfigurationManagementGraph's` configuration graph vertex. We create a configuration vertex for this graph by copying over the template's properties onto this graph's configuration vertex, and then "open" the graph according to that configuration.

The notion of "create" and "open" here is subtle; we are not referring to methods that "create" a graph and "open" a graph. We are referring to methods that 1. create a configuration for a graph based off of the already created template configuration and 2. open a graph based on an existing configuration vertex for that graph